builder/builder.go

```diff
@@ -443,3 +459,77 @@ func parseImageNameFromArgs(cmdArgs string) string {
 	}
 	return cmdArgs[:idx]
 }
+
+func (b *Builder) createFilesForVolume(ctx context.Context, volMount *volume.Volume) error {
```

Comment:
name this differently:

`processVolumeSource()`

Ideally, it should process the source according to source type:
```go
case source.Secret != nil:
            // base64 decode
            // dump content on to a file
case source.AzureFile != nil:   (future scenario)
           // download source code from remote
           // create a whole new file system
default:
    throw err("not supported yet") 
```

In this way, this method is more extensible and can process different volume source types differently.

Reply:
.. or `prepareVolumeSource()`